frame/pallet-xcm/src/tests.rs

```diff
@@ -0,0 +1,1248 @@
+            INITIAL_BALANCE - SEND_AMOUNT
+        );
+        assert_eq!(
+            sent_xcm(),
+            vec![(
+                RelayLocation::get().into(),
+                Xcm(vec![
+                    ReceiveTeleportedAsset((Here, SEND_AMOUNT).into()),
+                    ClearOrigin,
+                    buy_limited_execution((Here, SEND_AMOUNT), 5000),
+                    DepositAsset {
+                        assets: All.into(),
+                        max_assets: 1,
+                        beneficiary: dest
+                    },
+                ]),
+            )]
+        );
+        let versioned_sent = VersionedXcm::from(sent_xcm().into_iter().next().unwrap().1);
+        let _check_v0_ok: xcm::v0::Xcm<()> = versioned_sent.try_into().unwrap();
+        assert_eq!(
+            last_event(),
+            Event::XcmPallet(crate::Event::Attempted(Outcome::Complete(weight)))
+        );
+    });
+}
+
+/// Test `limited_teleport_assets` with unlimited weight
+///
+/// Asserts that the sender's balance is decreased as a result of execution of
+/// local effects.
+#[test]
+fn unlimmited_teleport_assets_works() {
+    let balances = vec![
+        (ALICE, INITIAL_BALANCE),
+        (ParaId::from(PARA_ID).into_account(), INITIAL_BALANCE),
+    ];
+    new_test_ext_with_balances(balances).execute_with(|| {
+        let weight = 2 * BaseXcmWeight::get();
+        assert_eq!(Balances::total_balance(&ALICE), INITIAL_BALANCE);
+        let dest: MultiLocation = AccountId32 {
+            network: Any,
+            id: BOB.into(),
+        }
+        .into();
+        assert_ok!(XcmPallet::limited_teleport_assets(
+            Origin::signed(ALICE),
+            Box::new(RelayLocation::get().into()),
+            Box::new(dest.clone().into()),
+            Box::new((Here, SEND_AMOUNT).into()),
+            0,
+            WeightLimit::Unlimited,
+        ));
+        assert_eq!(
+            Balances::total_balance(&ALICE),
+            INITIAL_BALANCE - SEND_AMOUNT
+        );
+        assert_eq!(
+            sent_xcm(),
+            vec![(
+                RelayLocation::get().into(),
+                Xcm(vec![
+                    ReceiveTeleportedAsset((Here, SEND_AMOUNT).into()),
+                    ClearOrigin,
+                    buy_execution((Here, SEND_AMOUNT)),
+                    DepositAsset {
+                        assets: All.into(),
+                        max_assets: 1,
+                        beneficiary: dest
+                    },
+                ]),
+            )]
+        );
+        assert_eq!(
+            last_event(),
+            Event::XcmPallet(crate::Event::Attempted(Outcome::Complete(weight)))
+        );
+    });
+}
+
+/// Test `reserve_transfer_assets`
+///
+/// Asserts that the sender's balance is decreased and the beneficiary's balance
+/// is increased. Verifies the correct message is sent and event is emitted.
+#[test]
+fn reserve_transfer_assets_works() {
```

Comment:
Please add UT for your `reserve_withdraw_assets`.

Reply:
Added